app/bundles/MessengerBundle/MessageHandler/EmailHitNotificationHandler.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\MessengerBundle\MessageHandler;
+
+use Doctrine\ORM\OptimisticLockException;
+use Mautic\EmailBundle\Model\EmailModel;
+use Mautic\MessengerBundle\Message\EmailHitNotification;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Exception\RecoverableMessageHandlingException;
+use Symfony\Component\Messenger\Handler\Acknowledger;
+use Symfony\Component\Messenger\Handler\MessageSubscriberInterface;
+
+class EmailHitNotificationHandler implements MessageSubscriberInterface
+{
+    public function __construct(private EmailModel $emailModel, private LoggerInterface $logger)
+    {
+    }
+
+    public function __invoke(EmailHitNotification $message, Acknowledger $ack = null): void
+    {
+        try {
+            $this->logger->debug('Processing email hit notification, statId '.$message->getStatId());
+            $this->emailModel->hitEmail(
+                $message->getStatId(),
+                $message->getRequest(),
+                false,
+                $message->isSynchronousRequest(),
+                $message->getEventTime(),
+                true
+            );
+        } catch (OptimisticLockException $lockException) {
```

Comment:
```suggestion
        } catch (\Doctrine\DBAL\Exception\RetryableException $lockException) {
```
I guess you wanted to catch **a deadlock exception** here ("Deadlock found when trying to get lock; try restarting transaction"). `OptimisticLockException` will never be thrown here as we don't use the optimistic locking anywhere in the codebase so far.